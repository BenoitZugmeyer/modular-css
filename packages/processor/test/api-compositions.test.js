"use strict";

const namer = require("@modular-css/test-utils/namer.js");

const Processor = require("../processor.js");

describe("/processor.js", () => {
    describe(".compositions", () => {
        let processor;

        beforeEach(() => {
            processor = new Processor({
                namer,
            });
        });

        it("should return compositions for loaded files", async () => {
            await processor.file(require.resolve("./specimens/start.css"));

            await expect(processor.compositions).resolves.toMatchSnapshot();
        });
    });
});
