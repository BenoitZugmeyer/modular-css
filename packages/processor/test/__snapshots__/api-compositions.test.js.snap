// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`/processor.js .compositions should return compositions for loaded files 1`] = `
Object {
  "packages/processor/test/specimens/folder/folder.css": Object {
    "folder": "folder",
  },
  "packages/processor/test/specimens/local.css": Object {
    "booga": "booga",
    "folder": "white",
    "looga": "booga looga",
    "one": "red",
    "two": "blue",
  },
  "packages/processor/test/specimens/start.css": Object {
    "booga": "booga",
    "folder": "white",
    "one": "red",
    "tooga": "tooga",
    "two": "blue",
    "wooga": "booga wooga",
  },
}
`;
